### PR TITLE
Add Machine Coeus to xml config files

### PR DESCRIPTION
**Pull request by colleenRooney**
Add changes necessary to run cesm on the machine coeus to xml config files.

Test suite: scripts_regression_tests.py
Test namelist changes: 
Test status: [2 failures](https://bb.cgd.ucar.edu/test-compiler-changeable-build-time)
Fixes N/A

User interface changes?: N

Update gh-pages html (Y/N)?: N

Code review:

### CODE REVIEW

File: config/cesm/machines/config_compilers.xml [hunk 1]
```diff
@@ -601,6 +601,60 @@ using a fortran linker.
   </SLIBS>
 </compiler>
 
+<compiler MACH="coeus" COMPILER="gnu">
```
Comment (jedwards4b) [inline config/cesm/machines/config_compilers.xml]: The idea in config_compilers.xml is to define a base set of flags for the gnu compiler (above in the file) and then only modify those settings for a particular machine as necessary.   Can you reduce this xml clause to be only the flags that are changed for coeus from the default gnu settings?

File: config/cesm/machines/config_machines.xml [hunk 2]
```diff
@@ -433,6 +433,63 @@ This allows using a different mpirun command to launch unit tests
...
+      Portland State University Coeus Cluster CentOS 7
+    </DESC>
+    <NODENAME_REGEX>(login[1,2].cluster|compute[0-9]*.cluster)</NODENAME_REGEX>
+    <OS>LINUX</OS>
+    <COMPILERS>gnu</COMPILERS>
+    <MPILIBS>mvapich2</MPILIBS>
+    <PROJECT>none</PROJECT>
+    <CIME_OUTPUT_ROOT>$ENV{CESMDATAROOT}/$USER</CIME_OUTPUT_ROOT>
+    <DIN_LOC_ROOT>$ENV{CESMDATAROOT}/inputdata</DIN_LOC_ROOT>
+    <DOUT_S_ROOT>$ENV{CESMDATAROOT}/archive/$CASE</DOUT_S_ROOT>
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml]: Usually you will want the DOUT_S_ROOT to be $USER specific like
$CIME_OUTPUT_ROOT/archive/$CASE

File: config/cesm/machines/config_machines.xml [hunk 3]
```diff
@@ -433,6 +433,63 @@ This allows using a different mpirun command to launch unit tests
...
+    <BASELINE_ROOT>$ENV{CESMDATAROOT}/cesm_baselines</BASELINE_ROOT>
+    <CCSM_CPRNC>/vol/apps/hpc/src/cesm-2.1.0/cime/tools/cprnc/cprnc</CCSM_CPRNC>
+    <GMAKE>make</GMAKE>
+    <GMAKE_J>8</GMAKE_J>
+    <BATCH_SYSTEM>slurm</BATCH_SYSTEM>
+    <SUPPORTED_BY>oit-rc-groups@pdx.edu</SUPPORTED_BY>
+    <MAX_TASKS_PER_NODE>19</MAX_TASKS_PER_NODE>
+    <MAX_MPITASKS_PER_NODE>19</MAX_MPITASKS_PER_NODE>
```
Comment (jedwards4b) [inline config/cesm/machines/config_machines.xml]: 19 is a very strange number for tasks per node - what type of machine is this?

Comment (colleenRooney) [reply config/cesm/machines/config_machines.xml]: Thank you for all your feedback. Our compute nodes are Dual Intel Xeon E2630 v4, 10 cores each. I have heard mixed advice on if this should be kept 1 below the number of cores, should I bump this up to 20?

File: config/cesm/machines/config_machines.xml (hunk 2 above)
Comment (colleenRooney) [reply config/cesm/machines/config_machines.xml]: That make sense I'll fix this now

File: config/cesm/machines/config_compilers.xml (hunk 1 above)
Comment (colleenRooney) [reply config/cesm/machines/config_compilers.xml]: I'll work on seeing what the minimum settings to still have the regression tests pass to the same level are.

File: config/cesm/machines/config_machines.xml (hunk 3 above)
Comment (jedwards4b) [reply config/cesm/machines/config_machines.xml]: There is occasionally good reason to leave a core available for the OS but we would typically set
MAX_MPI_TASKS_PER_NODE=20 and MAX_TASKS_PER_NODE=40 - (this allows for smt threads which may improve performance in some cases) Try it with a case or two and see what works best for you.  The PFS test is designed to measure performance of a given compset.

**Comment by colleenRooney** [comment]
@jedwards4b I have made all the suggested changes. All the tests in the regression tests script pass now. I have not used the PFS test to test performance of different max task settings yet, but I plan to tonight or tomorrow.

**Pull request merged**